### PR TITLE
add csv export option

### PR DESCRIPTION
**Pull request by d-monnet**

### CODE REVIEW

File: src/performance_profiles.jl [hunk 1]
```diff
@@ -152,3 +154,49 @@ function performance_profile(
...
+  max_elem = maximum(length.(x_data))
+  for i in eachindex(x_data)
+    append!(x_data[i],[NaN for i=1:max_elem-length(x_data[i])])
+    append!(y_data[i],[NaN for i=1:max_elem-length(y_data[i])])
+  end
+  x_mat = hcat(x_data...)
+  y_mat = hcat(y_data...)
+
+  isempty(solver_names) && (solver_names = ["solver_$i" for i = 1:nsolvers])
+  @show nsolvers solver_names
```
Comment (tmigot) [inline src/performance_profiles.jl:193]:
```suggestion
```

File: src/performance_profiles.jl [hunk 2]
```diff
@@ -152,3 +154,49 @@ function performance_profile(
...
+
+  header = vcat([[sname*"_x",sname*"_y"] for sname in solver_names]...)
```
Comment (tmigot) [inline src/performance_profiles.jl:195]: Could we add this to the kwargs?

File: src/performance_profiles.jl [hunk 3]
```diff
@@ -152,3 +154,49 @@ function performance_profile(
...
+
+* `T :: Matrix{Float64}`: each column of `T` defines the performance data for a solver (smaller is better).
+  Failures on a given problem are represented by a negative value, an infinite value, or `NaN`.
+* `filename :: String` : path to the export file.
+
+## Keyword Arguments
+
+* `solver_names :: Vector{S}` : names of the solvers
+
+Other keyword arguments are passed `performance_profile_data`.
```
Comment (tmigot) [inline src/performance_profiles.jl:173]:
```suggestion
Other keyword arguments are passed to `performance_profile_data`.
```

File: src/performance_profiles.jl [hunk 4]
```diff
@@ -152,3 +154,49 @@ function performance_profile(
     kwargs...,
   )
 end
+
+"""
+export_performance_profile(T; solver_names = [], logscale::Bool = true, sampletol=0.0, drawtol=0.0,)
```
Comment (tmigot) [inline src/performance_profiles.jl:159]:
```suggestion
  export_performance_profile(T; solver_names = [], logscale::Bool = true, sampletol=0.0, drawtol=0.0,)
```
And I don't think this signature match the function

File: src/performance_profiles.jl [hunk 5]
```diff
@@ -152,3 +154,49 @@ function performance_profile(
...
+
+## Arguments
+
+* `T :: Matrix{Float64}`: each column of `T` defines the performance data for a solver (smaller is better).
+  Failures on a given problem are represented by a negative value, an infinite value, or `NaN`.
+* `filename :: String` : path to the export file.
+
+## Keyword Arguments
+
+* `solver_names :: Vector{S}` : names of the solvers
```
Comment (tmigot) [inline src/performance_profiles.jl:171]:
```suggestion
* `solver_names :: Vector{S}` : names of the solvers.
```

**Review by tmigot — CHANGES_REQUESTED** [review]
Thanks @d-monnet ! It looks overall great, I just made some small comments.

File: src/performance_profiles.jl (hunk 2 above)
Comment (d-monnet) [reply src/performance_profiles.jl:194]: Do you mean the header ?

Comment (tmigot) [reply src/performance_profiles.jl:194]: Yes! So, that a user could personalize this without modifying the CSV file.

File: src/performance_profiles.jl [hunk 6]
```diff
@@ -152,3 +154,48 @@ function performance_profile(
     kwargs...,
   )
 end
+
+"""
+  export_performance_profile(T; solver_names = [], logscale::Bool = true, sampletol=0.0, drawtol=0.0,)
```
Comment (tmigot) [inline src/performance_profiles.jl:159]: this still doesn't match the function's signature below.

File: src/performance_profiles.jl (hunk 2 above)
Comment (d-monnet) [reply src/performance_profiles.jl]: It's done!

File: src/performance_profiles.jl (hunk 6 above)
Comment (d-monnet) [reply src/performance_profiles.jl]: It's solved.

File: test/runtests.jl [hunk 7]
```diff
@@ -61,4 +61,12 @@ if !Sys.isfreebsd() # GR_jll not available, so Plots won't install
     profile = data_profile(PGFPlotsXBackend(), H, T, labels)
     @test isa(profile, Plots.Plot)
   end
+
+  @testset "csv export" begin
+    T = 10 * rand(25, 3)
+    filename = "data.csv"
+    export_performance_profile(T,filename)
```
Comment (tmigot) [inline test/runtests.jl:68]: For the sake of keeping a good code coverage could you also add a test where the user provide another header?

**Review by tmigot — CHANGES_REQUESTED** [review]

**Comment by d-monnet** [comment]
I've just noticed that the last point of the profile computed by `perfomance_profile_data()` has y always coordinate `1.1`. I suppose this is useful for plot purpose, but I'm not sure we should export this point.

**Review by tmigot — APPROVED** [review]
Thank you @d-monnet it looks good to me

**Comment by tmigot** [comment]
> I've just noticed that the last point of the profile computed by `perfomance_profile_data()` has y always coordinate `1.1`. I suppose this is useful for plot purpose, but I'm not sure we should export this point.

Let me know if you want to add one last commit for this. I am also ok, keeping things like this, and eventually hopening an issue to explore

**Comment by d-monnet** [comment]
> > I've just noticed that the last point of the profile computed by `perfomance_profile_data()` has y always coordinate `1.1`. I suppose this is useful for plot purpose, but I'm not sure we should export this point.
> 
> Let me know if you want to add one last commit for this. I am also ok, keeping things like this, and eventually hopening an issue to explore

I wouldn't bring any more change. I think this extra coordinate point should be dealt with (if necessary) in `performance_profile_data` which creates it.

**Comment by d-monnet** [comment]
> Thank you @d-monnet it looks good to me

Thanks for your reviews ! I'll let you merge the PR if it's all good for you.

**Pull request merged**